lib/rex/ui/text/input/readline.rb

```diff
@@ -17,17 +17,18 @@ class Input::Readline < Rex::Ui::Text::Input
     #
     # Initializes the readline-aware Input instance for text.
     #
-    def initialize(tab_complete_proc = nil)
+    def initialize(tab_complete_proc)
       if(not Object.const_defined?('Readline'))
         require 'readline'
       end
 
       self.extend(::Readline)
 
-      if (tab_complete_proc)
+      if tab_complete_proc
         ::Readline.basic_word_break_characters = ""
-        ::Readline.completion_proc = tab_complete_proc
-        @rl_saved_proc = tab_complete_proc
+        ::Readline.completion_proc = with_error_handling(tab_complete_proc)
+
+        @rl_saved_proc = with_error_handling(tab_complete_proc)
```

Comment:
```suggestion
```suggestion
        @rl_saved_proc = with_error_handling(tab_complete_proc)
        ::Readline.completion_proc = @rl_saved_proc
```
```